cmd/kubeadm/app/preflight/checks.go

```diff
@@ -940,6 +945,7 @@ func RunJoinNodeChecks(execer utilsexec.Interface, cfg *kubeadmapi.JoinConfigura
 		DirAvailableCheck{Path: filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.ManifestsSubDirName)},
 		FileAvailableCheck{Path: filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.KubeletKubeConfigFileName)},
 		FileAvailableCheck{Path: filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.KubeletBootstrapKubeConfigFileName)},
+		SwapCheck{},
```

Comment:
this as to be wrapped for OS == linux only.

unfortunately we always run it on Join(), because we ignore the KubeletConfiguration on "join".
but this is out of scope for this patch.

Reply:
added OS == linux checks in both places. Thank you for pointing this out!